### ISSUE TITLE
Using @Composable reified inline functions fails to compile for iOS targets 

### ISSUE THREAD

**Issue by xxfast** [JetBrains/compose-multiplatform#2900]
**Describe the bug**
I have a simple `@Composable` inline function with reified template parameters like this 
```kotlin
@Composable
inline fun <reified C : Parcelable> rememberRouter(
  stack: List<C>,
  handleBackButton: Boolean = true
): Router<C> { ..}
```
compiling this for any ios target fails with
```
Compilation failed: Symbol for [ io.github.xxfast.decompose.router/rememberRouter|rememberRouter(kotlin.Any;kotlin.Boolean;kotlin.Function0<kotlin.collections.List<0:0>>){0§<kotlin.Any>}[0] <- Local[<TP>,0|TYPE_PARAMETER name:C index:0 variance: superTypes:[kotlin.Any] reified:false] ] is unbound
```
I think i have all my dependencies set correctly so not sure why this is popping up 🤔 

However, this error goes away as soon as you make the same function not inline and not reified 
```kotlin
@Composable
fun <C : Parcelable> rememberRouter(
  type: KClass<C>,
  stack: List<C>,
  handleBackButton: Boolean = true
): Router<C>
```

**Affected platforms**
Select one of the platforms below:
- iOS

**Versions**
- Kotlin version*: 1.9.20 
- Compose Multiplatform version*:  1.5.10

**To Reproduce**
Steps and/or the code snippet to reproduce the behavior:
1. Checkout branch from https://github.com/xxfast/Decompose-Router/pull/80
2. uncomment signature on line number 42-47
3. Open xcode project and hit run
4. Error on xcode build log on linking stage

**Expected behavior**
`@Composable` inline function with reified template parameters to build and linked correctly.

**Comment by pjBooms** [comment]
>Error on xcode build log on linking stage

Unfortunately, it fails earlier on my side with "unable to open configuration settings file".

Then I tried to copy-paste problematic code into https://github.com/JetBrains/compose-multiplatform-template
The compilation failed for `@Composable fun rememberRouter` for iOS target because it is declared public. It is known issue (you reported two different issues on this recently) that we are fixing now. Making the function `internal` fixes the problem even if I declare it as inline with reified generic parameter. Personally, I have no idea why the `rememberRouter` is compiled now on your side without declaring it `internal` but can you make it `internal` and `inline` with reified generic? We need to know if it is the same problem with `internal` or not?

**Comment by xxfast** [comment]
Thanks for the quick reply 🙂 unfortunately, I need this to be not internal for another multiplatform multimodule project where this `rememberRouter` exists in a `:core` module that each of the `:feature` modules consume. 

Feel free to close #2901 and #2903 if they are all caused by the same underlying issue. I'll watch this issue for future updates 👍

**Comment by xxfast** [comment]
With compose-multiplatform 1.5.10,. The error message is now updated to 
```
Compilation failed: Symbol for [ io.github.xxfast.decompose.router/rememberRouter|rememberRouter(kotlin.Any;kotlin.Boolean;kotlin.Function0<kotlin.collections.List<0:0>>){0§<kotlin.Any>}[0] <- Local[<TP>,0|TYPE_PARAMETER name:C index:0 variance: superTypes:[kotlin.Any] reified:false] ] is unbound
```

**Comment by bidrohi** [comment]
I'm seeming a similar error with version 1.6.10 with Circuit. Here's the call stack:

**Comment by MatkovIvan** [comment]
It sounds very similar to #3147 (fixed in 1.6.10)
cc @eymar @shishkin-pavel

**Comment by bidrohi** [comment]
> Hi, @bidrohi! could you please provide some reproducer?

I'm working to narrow down the issue to see if I can get to a small reproducable package.

**Comment by MatkovIvan** [comment]
Closing as solved in 1.6.10

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2900">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
